homework_solution.md

```diff
@@ -0,0 +1,474 @@
+# Part 1: Missing Station Data
+
+1. _**Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.**_
+
+   ### Steps
+
+   1. Create an index on `from_station_id` and `to_station_id` on `trips` table to help query on those
+
+      ```
+      CREATE INDEX idx_from_station ON trips USING btree (from_station_id);
+      ```
+
+      ```
+      CREATE INDEX idx_to_station ON trips USING btree (to_station_id);
+      ```
+
+   2. Delete the exist `stations` table and recreate it with serial ids, so we can rely on auto-increment. (We could try and alter the table, but that seems to be an involved process, and our table being currently empty, it's easier to just recreate it)
+
+      ```
+      DROP TABLE stations;
+      ```
+
+      ```
+      CREATE TABLE stations
+      (
+         id SERIAL,
+         name VARCHAR(100)
+      );
+      ```
+
+      The effect of this can be perceived by running `SELECT nextval(pg_get_serial_sequence('stations', 'id'));`.
+      On the old table, that sequence, though present, wouldn't return a value. Now it does.
+
+   3. Get the first name for stations, assuming that as the correct one where there's more than one, combining the `from` and `to` column results
```

Comment:
[question]
How could you further normalize the station names?  Why might you do this?

Reply:
Running a query using `SIMILAR TO`, we could further investigate cases where stations are sometimes named `Yonge / Dundas` and other cases as `Dundas / Yonge` (potentially having different ids).
Personally though, I don't see a way to automatically make decisions based on that query result, but rather see it as a way to aid in manual decisions and isolated steps in further sanitizing the data.

(Not sure if that's what you had in mind, so let me know if you were thinking of something else!)

Reply:
I was thinking more along the lines of the `LOWER()` or `UPPER()` functions - specify a canonical form for each station name. I'm not super clear on how SIMILAR TO works, but that sounds like a very interesting idea! If you have the time, I'd encourage you to explore it further.

The reason I had in mind is that [later on](https://github.com/talyh/bridge-bikeshare-exercise/pull/1/files/72b4806addd5a47ad809c6d830348de669f31f6f#r396119992), You're updating station ids in the `trips` table by comparing their station names to those in the `stations` table. If they're at all inconsistent, then there will be duplicate entries in the `stations` table (though with slightly different names), and then you'd have trips fromto the same station with different ids.  You explained your thinking well thee though, so I'm satisfied you know what I was shooting for :)